klaxer/sinks.py

```diff
@@ -114,16 +116,56 @@ def post_message(self, message):
             self.delete_message(last_message)
         return Message(**response)
 
+    def send_alert(self, alert):
+        last_message = self.get_last_message()
+        debounced = False
+        attachment = last_message.attachments[0] if last_message.attachments else None
+        if attachment and alert.message in unslack_text(attachment['text']): #TODO: Test more than just the message
+            debounced = True
+            alert.message = debounce(attachment['text'])
+        response = self.slack.chat.post_message(
+            channel=self.channel.id,
+            username=alert.username,
+            icon_emoji=alert.icon_emoji,
+            icon_url=alert.icon_url,
+            attachments=[{
+                'title': alert.title,
+                'text': alert.message,
+                'color': severity_to_color(alert.severity)
+            }]).body.get('message')
+        if debounced:
+            self.delete_message(last_message)
+        return Message(**response)
+
+def severity_to_color(severity):
+    """Map severity levels to colors"""
+    # These colors are from the Tomorrow Night Eighties colorscheme:
+    #  https://github.com/chriskempson/tomorrow-theme
+    if severity == Severity.CRITICAL:
+        return '#f2777a'
+    elif severity == Severity.WARNING:
+        return '#ffcc66'
+    elif severity == Severity.OK:
+        return '#99cc99'
+    elif severity == Severity.UNKNOWN:
+        return '#999999'
+    raise ValueError('Invalid severity supplied')
+
+def unslack_text(text):
+    """Slack applies formatting to inline URLs. This undoes it"""
+    has_url = URL_PATTERN.search(text)
+    if has_url:
+        # Searching only returns one match. Recursively unslack the message.
+        return unslack_text(text.replace(has_url.group(0), has_url.group('url')))
+    return text
 
 def debounce(text):
-    # Check for signs of a dup indicator
+    """Check for signs of a dup indicator, and increment the counter if present"""
```

Comment:
It would be nice to have an explicit test for this functionality, but as we've already said - we'll get to that.